flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsCheckpointStreamFactory.java

```diff
@@ -301,9 +301,16 @@ public StreamStateHandle closeAndGetHandle() throws IOException {
 					}
 					else {
 						flush();
+
+						long size = -1;
```

Comment:
I am not sure if returning -1 as size on exception is ideal. Currently, this value should one be used in the calculation of meta data, but one might be tempted to use it e.g. to preallocate a byte[] to read the file into, so this should at least be documented in `StateObject`. Furthermore, we make the assumption that the stream position is also equal to the final file size. Not entirely sure if this holds for all streams and file systems, but I guess this is the best we can do without asking the file system for meta data.